volume-providers/vpc/provider/util.go

```diff
@@ -110,43 +142,56 @@ func NewFlexyRetry(maxRtyAtmpt int, maxrRtyGap int) FlexyRetry {
 }
 
 // FlexyRetry ...
-func (fRetry *FlexyRetry) FlexyRetry(logger *zap.Logger, retryfunc func() (interface{}, error), skipRetryFunc func(interface{}, *models.Error) bool) error {
+func (fRetry *FlexyRetry) FlexyRetry(logger *zap.Logger, funcToRetry func() (error, bool)) error {
 	var err error
-
+	var stopRetry bool
 	for i := 0; i < fRetry.maxRetryAttempt; i++ {
 		if i > 0 {
 			time.Sleep(time.Duration(retryGap) * time.Second)
 		}
-		obj, err := retryfunc()
-		if err != nil {
-			//Skip retry for the below type of Errors
-			modelError, ok := err.(*models.Error)
-			if !ok {
-				continue
-			}
-			//! check if any custom skip needs to be done
-			if skipRetryFunc != nil {
-				if skipRetryFunc(obj, modelError) {
-					break
-				}
-			} else {
-				// Default as per provider
-				if skipRetry(modelError) {
-					break
-				}
-			}
-			if i >= 1 {
-				retryGap = 2 * retryGap
-				if retryGap > fRetry.maxRetryGap {
-					retryGap = fRetry.maxRetryGap
-				}
-			}
-			if (i + 1) < fRetry.maxRetryAttempt {
-				logger.Info("Error while executing the function. Re-attempting execution ..", zap.Int("attempt..", i+2), zap.Int("retry-gap", retryGap), zap.Int("max-retry-Attempts", maxRetryGap), zap.Error(err))
+		// Call function which required retry, retry is decided by funtion itself
+		err, stopRetry = funcToRetry()
+		if stopRetry {
+			break
+		}
+
+		// Update retry gap as per exponentioal
+		if i >= 1 {
+			retryGap = 2 * retryGap
+			if retryGap > fRetry.maxRetryGap {
+				retryGap = fRetry.maxRetryGap
 			}
-			continue
 		}
-		return err
+		if (i + 1) < fRetry.maxRetryAttempt {
+			logger.Info("UNEXPECTED RESULT, Re-attempting execution ..", zap.Int("attempt..", i+2),
+				zap.Int("retry-gap", retryGap), zap.Int("max-retry-Attempts", fRetry.maxRetryAttempt),
+				zap.Bool("stopRetry", stopRetry), zap.Error(err))
+		}
+	}
+	return err
+}
+
+// FlexyRetryWithConstGap ...
+func (fRetry *FlexyRetry) FlexyRetryWithConstGap(logger *zap.Logger, funcToRetry func() (error, bool)) error {
+	var err error
+	var stopRetry bool
+	// lets have more number of try for wait for attach and detach specially
+	totalAttempt := fRetry.maxRetryAttempt * 4 // 40 time as per default values i.e 400 seconds
```

Comment:
1. why multiplied by 4  ?  can't it  be found with   TOTALTIME_OUT/RETRY_GAP
totalAttempt := fRetry.maxRetryAttempt * 4 

2. Can this function be used for another function with a different TIMEOUT and different RETRY_GAP ? looks like Now its not reconfigurable....(say we want to use this with diffrent GAP for attach,detach,volumecreation, volumedeleteion) ?

Reply:
There is no TOTALTIME_OUT